client/landing/gutenboarding/gutenboard.tsx

```diff
@@ -57,14 +52,34 @@ export function Gutenboard() {
 	const [ isEditorSidebarOpened, updateIsEditorSidebarOpened ] = useState( false );
 	const toggleGeneralSidebar = () => updateIsEditorSidebarOpened( isOpen => ! isOpen );
 
-	const onboardingState = useSelect( select => select( STORE_KEY ).getState() );
+	const { siteTitle, siteVertical } = useSelect( select => select( STORE_KEY ).getState() );
+	const r = useRouteMatch( '*' );
+
+	let currentStep: Step;
+	let redirect: undefined | string;
+	let next: undefined | string;
+	let prev: undefined | string;
+	switch ( r?.url ) {
+		case '/':
+			currentStep = Step.IntentGathering;
+			if ( siteTitle ) {
+				next = '/design';
+			}
+			break;
 
-	// FIXME: Quick'n'dirty step state, replace with router
-	const [ currentStep, setStep ] = useState( Steps.IntentGathering );
-	const goToNextStep = stepCompleted[ currentStep ]( onboardingState )
-		? () => setStep( step => step + 1 )
-		: undefined;
-	const goToPrevStep = currentStep > 0 ? () => setStep( step => step - 1 ) : undefined;
+		case '/design':
+			currentStep = Step.DesignSelection;
+			if ( ! siteVertical ) {
+				redirect = '/';
+			}
+			prev = '/';
+			break;
+
+		default:
+			currentStep = Step.IntentGathering;
+			redirect = '/';
+			break;
+	}
```

Comment:
I'd kinda like to abstract this logic relatively early on :grimacing: While I definitely think that the current WP.com NUX flow suffers from over-engineering, I think we'll still want some kind of logic that defines e.g. that the user can only proceed to a step if some information has been specified in a previous step (hence my `stepCompleted` thingy) so that if people later add other steps, there's at least some framework to add a new step to.

Fine to leave as is for a first iteration I guess.

Reply:
I'm not attached to this bit, something like it was required to get routing working.

Our setup is a bit odd because all routes mount the same component (block) and routing is effectively a side-effect (dispatch action to update block attributes). I don't think "normal" react-router usage follows that pattern.

Reply:
> I'm not attached to this bit, something like it was required to get routing working.
> 
> Our setup is a bit odd because all routes mount the same component (block) and routing is effectively a side-effect (dispatch action to update block attributes). I don't think "normal" react-router usage follows that pattern.

Depends. The block might be peculiar, but I think a lot of apps that use react-router will have some top-level chrome that's present on pretty much all routes. So maybe not that exotic after all.